### PR TITLE
chore: merge build matrix set-up back to dsaidgovsg

### PR DESCRIPTION
**Pull request by guangie88**
Main changes include:

- Switch from Travis CI to GitHub Actions
- Performs a wide matrix of build, with `airflow` version being first
  priority and being offered both 1.9 and 1.10
- Docker images are tagged starting from `v1` until `v7`. Each version
  number change represents breaking changes to the image (just like an
  executable) and changes are detailed in
  [`CHANGELOG.md`](CHANGELOG.md), whose changes are typically of the
  following:
  - New required env vars added
  - Underlying Linux distro is changed
  - Dependent framework used changed (e.g. `conda` -> `poetry`)

### CODE REVIEW

**Comment by guangie88** [comment]
This is basically a merge from https://github.com/guangie88/airflow-pipeline/tree/master into `master` here with deconflict.

EDIT: To be more specific, the commit SHA is this: https://github.com/guangie88/airflow-pipeline/commit/6e464e61832c77a5b32c5a35dcd4c390da52a83c

File: entrypoint.sh [hunk 1]
```diff
@@ -1,87 +1,112 @@
...
+  echo "Admin user added!"
+fi
+
+# We assume the the patch/Z version is at least 11, based on the current edit
+# Thus it will definitely have both the RBAC and create_user features
+AIRFLOW_VERSION="$(airflow version)"
+AIRFLOW_Y_VERSION="$(echo ${AIRFLOW_VERSION} | cut -d . -f 2)"
+
+# Requires 'rbac' mode to be set to true to run the command properly
+if check_set "${ENABLE_AIRFLOW_RBAC_SETUP_AUTH}" && [ "${AIRFLOW_Y_VERSION}" -eq 10 ]; then
```
Comment (fred4freedom) [inline entrypoint.sh:89]: may need to update for version 2

**Review by fred4freedom — APPROVED** [review]
lgtm. some portions assume airflow consist of only version 1. may have to update in the future to prevent unwittingly breakage

File: entrypoint.sh (hunk 1 above)
Comment (guangie88) [reply entrypoint.sh:89]: Yup thanks for highlighting this!

**Pull request merged**